### PR TITLE
Allow to input timebar days and years with keyboard

### PR DESCRIPTION
**Pull request by j8seangel**
Modify the DateSelector component to allow enter dates with the keyboard for days and years.
Use native select for months.

### CODE REVIEW

File: src/timebar/components/date-selector.js [hunk 1]
```diff
@@ -1,32 +1,97 @@
...
 class DateSelector extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      value: props.value,
+      error: false,
+    }
+  }
+
+  componentWillReceiveProps(nextProps) {
```
Comment (j8seangel) [inline src/timebar/components/date-selector.js]: It is deprecated and not a good practice, but was the easiest way to reuse the existing code.

File: src/timebar/components/date-selector.js [hunk 2]
```diff
@@ -1,32 +1,121 @@
...
 import { ReactComponent as IconArrowDown } from '../icons/arrowDown.svg'
 
 class DateSelector extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      value: props.value,
+      error: false,
+    }
+    this.months = Array.from(Array(12).keys()).map((i) => ({
```
Comment (nerik) [inline src/timebar/components/date-selector.js]: Nice!

File: src/timebar/components/date-selector.js [hunk 3]
```diff
@@ -1,32 +1,97 @@
...
+      this.setState({ value: nextProps.value })
+    }
+  }
+
+  validate(value) {
+    const { unit } = this.props
+    const isNumber = typeof value === 'number'
+    if (!isNumber) return false
+
+    // TODO: validate days / months / years
```
Comment (nerik) [inline src/timebar/components/date-selector.js]: I think you could get away with just using https://github.com/iamkun/dayjs/blob/dev/docs/en/API-reference.md#validation-isvalid
You'd need to move the validation step to onStartChange/onEndChange in TimerangeSelector

Comment (j8seangel) [reply src/timebar/components/date-selector.js]: Good catch, now it has a better validation as also check if the day is included in the current month selected

**Comment by nerik** [comment]
Closed in favor of https://github.com/GlobalFishingWatch/map-components/pull/55

**Pull request closed without merging**